chain/starknet/src/data_source.rs

```diff
@@ -53,6 +60,21 @@ pub struct MappingBlockHandler {
     pub handler: String,
 }
 
+#[derive(Clone, Deserialize)]
+pub struct MappingEventHandler {
+    pub handler: String,
+    pub event: String,
```

Comment:
Defining `event` here as `String` for the resolved type `MappingEventHandler` is problematic. Evey time we need to check for matches the `event` field has to be encoded into selector (with the `key()` function from below), this is wasteful and has the unwanted effect of having to deal with error handling at index-time.

The existing codebase already deals with the parsing (and any pre-processing) or raw handler metadata with the `UnresolvedDataSource::resolve()` method. Any preprocessing logic should live there and surface the errors at resolve time.